node/node_handler.go

```diff
@@ -441,7 +460,12 @@ func (node *Node) pongMessageHandler(msgPayload []byte) int {
 	node.Consensus.Leader.PubKey = &bls.PublicKey{}
 	err = node.Consensus.Leader.PubKey.Deserialize(pong.LeaderPubKey)
 	if err != nil {
-		utils.GetLogInstance().Error("Unmarshal Leader PubKey Failed", "error", err)
+		utils.GetLogInstance().Error("Unmarshal Consensus Leader PubKey Failed", "error", err)
+	}
+	node.DRand.Leader.PubKey = &bls.PublicKey{}
```

Comment:
Why don't we introduce and use `func (drand *DRand) SetLeaderKey(key *bls.PublicKey)`?  This way, we don't have to flip the private `leader` to public, and if needed, the `SetLeaderKey()` would be a point of sanity check, e.g. resetting an already set leader key could be logged (since it should not happen normally IINM).

Reply:
I'll do it in the following PR.